inloop/solutions/validators.py

```diff
@@ -0,0 +1,45 @@
+from django.core.exceptions import ValidationError
+from django.dispatch import receiver
+
+from constance import config
+from constance.signals import config_updated
+
+
+def get_allowed_file_endings():
+    """
+    Return all allowed file endings.
+    """
+    return [ending.strip().lower() for ending in config.ALLOWED_FILE_ENDINGS.split(",")]
+
+
+allowed_file_endings = get_allowed_file_endings()
+
+
+def validate_filenames(upload):
+    """
+    Verify that all uploaded files comply to the given naming constraints.
+    """
+    error = ValidationError(
+        "One or more files contain disallowed file endings. "
+        "(Allowed: {})".format(", ".join(allowed_file_endings))
+    )
+    if not upload:
+        raise error
+    for file_name, _ in upload.items():
+        lower_file_name = file_name.lower()
+        for ending in allowed_file_endings:
+            if lower_file_name.endswith(ending):
+                break
+        else:
+            raise error
+
+
+@receiver(config_updated, dispatch_uid="general_config_updated")
+def handle_config_updated(sender, key, **kwargs):
+    """
+    Update allowed file endings on change.
+    """
+    global allowed_file_endings
```

Comment:
Bad style (avoid `global` whenever possible). I get the idea behind this, but it should be done another way.